cocotb/drivers/amba.py

```diff
@@ -146,11 +144,11 @@ def write(self, address, value, byte_enable=0xf, address_latency=0,
 
         # Wait for the response
         while True:
-            yield ReadOnly()
             if self.bus.BVALID.value and self.bus.BREADY.value:
                 result = self.bus.BRESP.value
                 break
             yield RisingEdge(self.clock)
+            yield ReadOnly()
```

Comment:
This change doesn't look right to me - we shouldn't be looking at signals until the `ReadOnly` phase